src/gadgets/point_addition/add.rs

```diff
@@ -1,14 +1,113 @@
-use bulletproofs::r1cs::{ConstraintSystem, LinearCombination as LC, Variable};
-use zerocaf::edwards::{AffinePoint, EdwardsPoint};
-use zerocaf::field::FieldElement as Fq;
-use zerocaf::ristretto::RistrettoPoint;
+use crate::helpers::{fq_as_scalar, n_point_coords_to_LC};
+use bulletproofs::r1cs::{
+    ConstraintSystem, LinearCombination as LC, Prover, R1CSError, R1CSProof, Variable, Verifier,
+};
+use bulletproofs::{BulletproofGens, PedersenGens};
+use curve25519_dalek::scalar::Scalar;
+use merlin::Transcript;
+use rand::thread_rng;
+use zerocaf::field::FieldElement;
+use zerocaf::ristretto::{CompressedRistretto, RistrettoPoint};
 
+/// Builds a proof which holds the constraints related to
+/// the point addition of two publicly known RistrettoPoints.
+pub fn point_addition_proof(
+    pc_gens: &PedersenGens,
+    bp_gens: &BulletproofGens,
+    p1: RistrettoPoint,
+    p2: RistrettoPoint,
+    p3: RistrettoPoint,
+    a: FieldElement,
+    d: FieldElement,
+) -> Result<R1CSProof, R1CSError> {
```

Comment:
I believe Point addition does not require a secret scalar, but Scalar multiplication does

Reply:
This is the point addition gadget. As stated in #5 this is only one of the components of the Sk knowledge gadget. The `Scalar` will appear there and on the conditionally selection part but has nothing to do with the Point addition or point doubling as Kev stated.

Here, since the Pk and the Basepoint are publicly known, you don't need to commit them.